core/include/cubos/core/reflection/traits/fields.hpp

```diff
@@ -0,0 +1,146 @@
+/// @file
+/// @brief Class @ref cubos::core::reflection::FieldsTrait.
+/// @ingroup core-reflection
+
+#pragma once
+
+#include <string>
+
+#include <cubos/core/reflection/reflect.hpp>
+
+namespace cubos::core::reflection
+{
+    /// @brief Describes the fields of a reflected type.
+    /// @see See @ref examples-core-reflection-traits-fields for an example of using this trait.
+    /// @ingroup core-reflection
+    class FieldsTrait
+    {
+    public:
+        /// @brief Interface for obtaining the address of a field on a given instance.
+        /// @note We could use std::function and a capturing lambda instead of this interface, but
+        /// that would require including the bloated `<functional>` header here, which we want to
+        /// avoid.
```

Comment:
Gotta love finding STL rewrites in the wild, we should just commit to #560 and extract these to the same place instead of sprinkling them over the codebase

Reply:
Open an issue please

Reply:
The task is enough, you can merge